src/cogent3/app/align.py

```diff
@@ -129,13 +129,13 @@ def align_to_named_seq(self, seqs):
             # eliminating positions with deletions in the reference
             result = result.filtered(no_ref_gap)
             if aligned is None:
-                aligned = result.to_type(array_align=False)
+                aligned = result.to_moltype(self._moltype)
                 continue
 
-            aligned = aligned.add_from_ref_aln(result.to_type(array_align=False))
```

Comment:
as the `add_from_ref_aln` method already does the type conversion when required, change this to
```python
aligned = aligned.add_from_ref_aln(result)
```